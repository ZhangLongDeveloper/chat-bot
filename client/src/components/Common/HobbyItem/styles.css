.hobby-container {
  margin: 20px 0;
}

.hobby-container .image {
  width: 100%;
  max-width: 150px;
  margin: auto;
}

.hobby-container .name {
  margin: 0;
}

.hobby-container .description {
  text-align: justify;
  text-indent: 15px;
}

@media (max-width: 750px) {
  
}

@media (max-width: 600px) {
  .hobby-container {
    text-align: center;
  }
}