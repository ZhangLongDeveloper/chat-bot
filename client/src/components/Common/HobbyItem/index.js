import React from 'react';
import PropTypes from 'prop-types';

import './styles.css';

const HobbyItem = ({ image, name, description }) => (
  <div className="hobby-container row valign-wrapper">
    <div className="col s12 m3 visual">
      <h5 className="name">{name}</h5>
      <img src={image} alt="icon" className="image" />
    </div>
    <div className="col s12 m9">
      <p className="description">{description}</p>
    </div>
  </div>
);

HobbyItem.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default HobbyItem;
