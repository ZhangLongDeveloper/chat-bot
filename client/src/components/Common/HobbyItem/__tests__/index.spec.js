import React from 'react';
import { create } from 'react-test-renderer';

import HobbyItem from '../../HobbyItem';

const mockComponent = props => (
  <HobbyItem {...props} />
);

describe('components/Common/<HobbyItem />', () => {
  it('should render component', () => {
    const props = {
      data: {
        name: 'name',
        image: 'image',
        description: 'description',
      },
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
