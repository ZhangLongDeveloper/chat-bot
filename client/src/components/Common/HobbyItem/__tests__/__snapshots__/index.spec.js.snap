// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Common/<HobbyItem /> should render component 1`] = `
<div
  className="hobby-container row valign-wrapper"
>
  <div
    className="col s12 m3 visual"
  >
    <h5
      className="name"
    />
    <img
      alt="icon"
      className="image"
    />
  </div>
  <div
    className="col s12 m9"
  >
    <p
      className="description"
    />
  </div>
</div>
`;
